docs/modules/serialization/pages/compact-serialization.adoc

```diff
@@ -35,368 +35,1506 @@ a class to implement a particular interface. Serializers might be implemented an
+
+    @Override
+    public String getTypeName() {
+        return "employee";
+    }
+}
+----
+--
+
+.NET::
++
+--
+[source,cs]
+----
+
+----
+--
+
+C++::
++
+--
+[source,cpp]
+----
+template<>
+struct hz_serializer<employee> : public compact_serializer
+{
+    static employee read(compact_reader& reader)
+    {
+        employee e;
+        
+        e.id = reader.read_int64("id");
+        e.name = reader.read_string("name");
+        
+        return e;
+    }
+
+    static void write(const employee& e, compact_writer& writer)
+    {
+        writer.write_int64("id", e.id);
+        writer.write_string("name", e.name);
+    } 
+
+    static std::string type_name() { return "employee"; }    
+};
+----
+--
+
+Node.js::
++
+--
+[source,javascript]
+----
+class EmployeeSerializer extends CompactSerializer {
+    read(reader) {
+        const id = reader.readInt64('id');
+        const name = reader.readString('name');
+        return new Employee(id, name);
+    }
+
+    write(writer, employee) {
+        writer.writeInt64('id', employee.id);
+        writer.writeString('name', employee.name);
+    }
+
+    getClass() {
+        return Employee;
+    }
+
+    getTypeName() {
+        return 'employee';
+    }
+};
+----
+--
+
+Python::
++
+--
+[source,python]
+----
+class EmployeeSerializer(CompactSerializer[Employee]):
+    def read(self, reader: CompactReader):
+        id = reader.read_int64("id")
+        name = reader.read_string("name")
+        return Employee(id, name)
+
+    def write(self, writer: CompactWriter, employee: Employee):
+        writer.write_int64("id", employee.id)
+        writer.write_string("name", employee.name)
+
+    def get_type_name(self):
+        return "employee"
+
+    def get_class(self):
+        return Employee
+----
+--
+
+Go::
++
+--
+[source,go]
+----
+
+----
+--
+
+====
+
+The last step is to register the serializer in the member or client configuration, as shown
+in the <<configuration>> section.
+
+Upon serialization, a schema will be created from the serializer, and a unique schema identifier
+will be assigned to it automatically.
+
+After the configuration registration, Hazelcast will serialize instances of the `Employee`
+class using the `EmployeeSerializer`.
+
+== Supported Types
+
+Compact serialization supports the following list as first class types. Any other type
+can be implemented on top of these, by using these types as building blocks.
+
+[cols="1m,1a,1a,1a,1a,1a,1a,1a]
+|===
+| Type | Java | .NET | C++ | Node.js | Python | Go | Description
+
+| BOOLEAN
+| boolean
+|
+| bool
+| boolean
+| bool
+|
+| True or false represented by a single bit as either 1 or 0.
+Up to eight booleans are packed into a single byte.
+
+| ARRAY_OF_BOOLEAN
+| boolean[]
+|
+| boost::optional<std::vector<bool>>
+| boolean[]
+| Optional[list[bool]]
+|
+| Array of booleans or null. Up to eight boolean array items are packed into a single byte.
+
+| NULLABLE_BOOLEAN
+| Boolean
+| 
+| boost::optional<bool>
+| boolean
```

Comment:
@harunalpak I think this should be `boolean | null` for Node.js. The same applies to others as well. Can we make sure that the type mappings here are the same as our API at here https://github.com/hazelcast/hazelcast-nodejs-client/blob/master/src/serialization/compact/CompactWriter.ts?

Reply:
Fixed